src/actions/assign.ts

```diff
@@ -0,0 +1,116 @@
+import { Context, Octokit } from "probot"
+import { different } from "../utils"
+
+async function assign(context: Context) {
+  const PRnumber = context.payload.number
+  const owner = context.payload.pull_request.head.user.login
+  const repo = context.payload.pull_request.head.repo.name
+
+  context.log.info("PR number:", PRnumber, ", owner:", owner, ", repo:", repo)
+
+  const oldReviewers: string[] = context.payload.pull_request.requested_reviewers.map(
+    (reviewer: { login: string }) => reviewer
+  )
+
+  let newReviewers: string[] = []
+  try {
+    newReviewers = await reviewersOfPR(context, PRnumber, owner, repo)
+  } catch (err) {
+    context.log.error(err, "Error while calling reviewersOfPR function")
+    return
+  }
+
+  // Check if old and new reviewers differ. Otherwise we can skip an API call
+  if (different(oldReviewers, newReviewers)) {
+    // Log reviewers replacement
+    context.log.info("Old reviewers:", oldReviewers)
+    context.log.info("New reviewers:", newReviewers)
+
+    const params = context.repo({ pull_number: PRnumber, reviewers: newReviewers })
+    const response = await context.github.pulls.createReviewRequest(params)
+    context.log.info("Status:", response.headers.status)
+  } else {
+    context.log.info("No reviewers changed")
+  }
+}
+
+interface CodeOwner {
+  path: string
+  username: string
+}
+
+function parseModifiedFiles(filesResponse: Octokit.PullsListFilesResponse): string[] {
+  const modifiedFiles: string[] = []
+
+  for (const file of filesResponse) {
+    modifiedFiles.push(file.filename)
+  }
+
+  return modifiedFiles
+}
+
+function parseCodeOwners(codeownersFile: string): CodeOwner[] {
+  const codeOwners: CodeOwner[] = []
+  for (const line of codeownersFile.split("\n")) {
+    const [path, owner] = line.split(" ")
+
+    // Ignore first line and empty newline at the end of the file
+    if (!owner?.includes("@")) {
```

Comment:
A CODEOWNERS line might have the owner field in the form of an email, but since we're diffing reviewers using their `login` field (i.e. username), we need to only allow `@username` forms, so let's use `.startWith()` instead.

Reply:
: )